plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/task/TaskManagerImpl.java

```diff
@@ -325,9 +335,13 @@ public DBTTaskRunStatus runTask(@NotNull DBRProgressMonitor monitor, @NotNull DB
     public TaskRunJob scheduleTask(@NotNull DBTTask task, @NotNull DBTTaskExecutionListener listener) {
         final TaskRunJob runJob = createJob((TaskImpl) task, listener);
         runJob.schedule();
+        if (serviceJob == null) {
```

Comment:
The `runningTasks.add(job);` is missing here. Is this expected?

Reply:
The  management of `runningTasks` goes from job listener, method `abouteToRun`